File: ch2_labs/lab4_C#/FigureLibrary/Rhombus.cs
namespace FigureLibrary
{
    public class Rhombus : Figure
    {
        public Rhombus(Point[] points) : base(points) { }

        public double GetPerimeter()
        {
            double side = SideLength(0);
            return 4 * side;
        }

        public double GetArea()
        {
            double d1 = vertices[0].DistanceTo(vertices[2]);
            double d2 = vertices[1].DistanceTo(vertices[3]);
            return (d1 * d2) / 2;
        }

        public string GetInfo()
        {
            return $"Rhombus with vertices: " +
                   $"{string.Join(", ", vertices.Select(p => $"({p.X}, {p.Y})"))}";
        }
    }
}

File: ch2_labs/lab4_C++/FigureLibrary/Rhombus.cpp
#include "Rhombus.h"

Rhombus::Rhombus(const std::vector<Point>& points) : Figure(points) {}

double Rhombus::getPerimeter() const {
    double side = getSideLength(0);
    return 4 * side;
}

double Rhombus::getArea() const {
    double d1 = vertices[0].distanceTo(vertices[2]);
    double d2 = vertices[1].distanceTo(vertices[3]);
    return (d1 * d2) / 2;
}
